addons/controllers/addon_controller.go

```diff
@@ -56,6 +56,7 @@ type AddonReconciler struct {
 
 // SetupWithManager performs the setup actions for an add on controller, using the passed in mgr.
 func (r *AddonReconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager, options controller.Options) error {
+	// nolint:dupl
```

Comment:
I had used the lint suppression because linter was complaining of duplicate code for all 3 Watches in clusterbootstrap and addoncontroller as below:

```
controllers/clusterbootstrap_controller.go:77: 77-100 lines are duplicate of `controllers/addon_controller.go:59-82` (dupl)
	ctrlr, err := ctrl.NewControllerManagedBy(mgr).
		For(&clusterapiv1beta1.Cluster{}).
		Watches(
			&source.Kind{Type: &runtanzuv1alpha3.TanzuKubernetesRelease{}},
			handler.EnqueueRequestsFromMapFunc(r.TKRToClusters),
			builder.WithPredicates(
				predicates.TKR(r.Log),
			),
		).
		Watches(
			&source.Kind{Type: &runtanzuv1alpha3.ClusterBootstrap{}},
			handler.EnqueueRequestsFromMapFunc(r.ClusterBootstrapToClusters),
			builder.WithPredicates(
				predicates.TKR(r.Log),
			),
		).
		Watches(
			&source.Kind{Type: &corev1.Secret{}},
			handler.EnqueueRequestsFromMapFunc(r.SecretsToClusters),
			builder.WithPredicates(
				predicates.TKR(r.Log),
			),
		).
		WithOptions(options).
```

Reply:
Fixed